extensions/axon/runtime/src/main/java/io/quarkus/axon/runtime/AxonClientProducer.java

```diff
@@ -0,0 +1,191 @@
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.SimpleCommandBus;
+import org.axonframework.commandhandling.distributed.AnnotationRoutingStrategy;
+import org.axonframework.commandhandling.gateway.CommandGateway;
+import org.axonframework.commandhandling.gateway.DefaultCommandGateway;
+import org.axonframework.config.Configuration;
+import org.axonframework.config.Configurer;
+import org.axonframework.config.DefaultConfigurer;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.queryhandling.*;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.serialization.upcasting.event.NoOpEventUpcaster;
+import org.axonframework.serialization.xml.XStreamSerializer;
+
+@ApplicationScoped
+public class AxonClientProducer {
+
+    private AxonRuntimeConfig axonRuntimeConfig;
+
+    private List<Object> sagas = new ArrayList<>();
+    private List<Object> aggregates = new ArrayList<>();
+    private List<Object> commandHandlers = new ArrayList<>();
+    private List<Object> eventHandlers = new ArrayList<>();
+    private List<Object> queryHandlers = new ArrayList<>();
+
+    void setAxonRuntimeConfig(AxonRuntimeConfig axonRuntimeConfig) {
+        this.axonRuntimeConfig = axonRuntimeConfig;
+    }
+
+    @Produces
+    @ApplicationScoped
+    public Configuration initializeConfiguration(Configurer configurer) {
+        Configuration configuration = configurer.buildConfiguration();
+        if (axonRuntimeConfig.autostart) {
+            configuration.start();
+        }
+        return configuration;
+    }
+
+    @Produces
+    @Dependent
+    public Configurer quarkusDefaultConfigurer(EventBus eventBus, CommandBus commandBus, QueryBus queryBus,
+            Serializer serializer) {
+
+        Configurer configurer = DefaultConfigurer.defaultConfiguration(false)
+                .configureEventBus(c -> eventBus)
+                .configureCommandBus(c -> commandBus)
+                .configureQueryBus(c -> queryBus)
+                .configureSerializer(c -> serializer);
+
+        // When registering a Saga or Aggregate also the EventSourcingHandlers inside
+        // of that class are registered
+        sagas.forEach(saga -> configurer.eventProcessing().registerSaga(saga.getClass()));
+        aggregates.forEach(aggregate -> configurer.configureAggregate(aggregate.getClass()));
+
+        // Command handlers, Event handlers and Query handlers must be registered by itself.
+        commandHandlers.forEach(commandHandler -> configurer.registerCommandHandler(conf -> commandHandler));
+        eventHandlers.forEach(eventHandler -> configurer.registerEventHandler(conf -> eventHandler));
+        queryHandlers.forEach(queryHandler -> configurer.registerQueryHandler(conf -> queryHandler));
+
+        return configurer;
+    }
+
+    @Produces
+    @Dependent
+    public AxonServerConfiguration axonServerConfiguration() {
```

Comment:
I see that `AxonServerConfiguration` uses `@ConfigurationProperties` from Spring Boot. Can you explain what the relationship of Axon with Spring Boot is?